vignettes/GMAT-GMST.Rmd

```diff
@@ -0,0 +1,126 @@
+
+This vignette demonstrates how to set up the calculation between GMAT and GMST and how to plot results to observe differences over time. 
+
+## Setup
+
+Let's load the packages we will need and run a scenario.
+
+``` {r hector.init}
+library(hector)
+library(ggplot2)
+
+inifile <- file.path(system.file('input', package = 'hector'),
+                     'hector_rcp45.ini')
+core <- newcore(inifile)
+run(core)
+```
+
+
+## Find GMST
+
+GMST can be calculated from Hector outputs by computing a weighted sum of the land air and ocean surface temperatures using the following equation:
+
+$ GMST = (land air temperature * fractional land area) + (ocean surface temperature * (1 - fractional land area)) $
+
+In Hector, land air temperature is represented by `LAND_AIR_TEMP()` and `Tgav_land` while ocean surface temperature by `OCEAN_SURFACE_TEMP()` and `Tgav_ocean_ST`. These components stem from the variable `Tgav`, which is how Hector stores GMAT (along with `GLOBAL_TEMP()`.)
+
+``` {r find.GMST}
+flnd = 0.29;     # fractional land area, a constant in Hector
+
+hector_output <- fetchvars(core, 1745:2300, vars = c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP(), GLOBAL_TEMP()))
+
+# Get the variables we want and corresponding years/value
+output <- subset(hector_output, variable %in% c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP()))
+output <- subset(output, select = c(year, variable, value))
+
+# Reorder data
+output <- reshape(output, direction = "wide", idvar = "year", timevar = "variable")
+names(output) <- gsub("value.", "", names(output))
+
+# Conversion equation
+output$Tgav_land <- output$Tgav_land * flnd
+output$Tgav_ocean_ST <- output$Tgav_ocean_ST * (1 - flnd)
+global_temp <- output$Tgav_land + output$Tgav_ocean_ST
+global_temp <- as.data.frame(global_temp) 
+
+# Organize final data, year and corresponding GMST
+global_surface_temp <- cbind(output$year, global_temp)
+names(global_surface_temp)[names(global_surface_temp) == "output$year"] <- "year"
+names(global_surface_temp)[names(global_surface_temp) == "global_temp"] <- "value"
+
+variable <- rep("GMST", times = 556)
+variable <- as.data.frame(variable)
+global_surface_temp <- cbind(global_surface_temp, variable)
+
+
+# Rename Hector's variable for GMAT
+global_air_temp <- subset(hector_output, variable == "Tgav")
+global_air_temp <- subset(global_air_temp, select = c(variable, year, value))
+global_air_temp$variable <- "GMAT"
+
+```
+
+## Plot 
+
+The following plot shows GMST and GMAT over time. We see that recently, GMAT is higher than GMST as GMAT is responding more quickly to anthropogenic forcing than GMST. This makes sense as GMST accounts for heating over the oceans which warm more slowly than the atmosphere. 
```

Comment:
Question about "We see that recently" are you referring to a point in time of the Hector run (the historical period, near present-day, near future, or long term future) or are you using this phrase to introduce the idea that our recent results out of Hector show, x, y, and z?

Reply:
Thanks for pointing this out, I forgot that the graph extends to 2300, not present day. I'll clarify this.

Reply:
> I think that "Tgav_land (LAND_AIR_TEMP()) makes more sense, clarity-wise. For my own understanding, why _are_ there two different variable names?

@bpbond please expand/clarify as needed. 

So "Tgav_land" is a string message that is passed around in Cpp, `LAND_AIR_TEMP()` is the function in R that returns this string which we pass to the underlying Cpp Hector model. When the R Hector package was constructed I belive the idea was that using an R function to generate the properly formatted Cpp string would a) be easier on users (users wouldn't have to worry about spelling/capitalization) and b) ease of development, we could change the string in Cpp however needed without changing the user interface. 

My guess as to why the R function is called LAND_AIR_TEMP instead of Tgav_land is because land air temp is more widely recognized than Tgav_land.

Reply:
I don't really have anything to add. Well explained @kdorheim .

Reply:
Should I add in a note about the Cpp message vs R function in the vignette? @bpbond

Reply:
Yes please; I think that would be helpful.